lib/katello/tasks/update_content_default_http_proxy.rake

```diff
@@ -0,0 +1,59 @@
+require File.expand_path("../engine", File.dirname(__FILE__))
+
+namespace :katello do
+  desc "Sets the content default http proxy to an existing http proxy based on supplied URL."
+  task :update_default_http_proxy => :environment do
+    setting = ::Setting::Content.where(name: 'content_default_http_proxy').first
+    options = {}
+    o = OptionParser.new
+    o.banner = "Usage: rake katello:update_content_default_http_proxy [options]"
+    o.on("-n", "--name HTTP_PROXY_NAME") do |name|
+      options[:name] = name
+    end
+    o.on("-u", "--url HTTP_PROXY_URL") do |url|
+      options[:url] = url
+    end
+    o.on("-h, --help", "Prints this help") do
+      puts o
+      exit
+    end
+    ordered_args = o.order!(ARGV) {}
+    o.parse!(ordered_args)
+
+    unless options.key?(:name)
+      $stderr.print("ERROR: Missing required option for --name HTTP_PROXY_NAME")
+      exit 2
+    end
+
+    User.current = User.anonymous_api_admin
+    http_proxy = HttpProxy.where(name: options[:name]).first
+
+    if http_proxy
+      setting.update_attribute(:value, http_proxy.name)
+      puts "Content default http proxy set to #{http_proxy.name_and_url}."
+    else
+      if options.key?(:url)
+        uri = URI(options[:url])
+        username = uri.user
+        password = uri.password
+        new_proxy = ::HttpProxy.new(name: options[:name], url: options[:url],
+                                  username: username, password: password,
+                                  organizations: Organization.all,
+                                  locations: Location.all)
+        if new_proxy.save!
+          setting.update_attribute(:value, new_proxy.name)
+          puts "Default content http proxy set to \"#{new_proxy.name_and_url}\"."
+        end
+      else
+        $stderr.print("ERROR: No http proxy found with name \"#{options[:name]}\".")
```

Comment:
this error doesn't make sense for this condition does it?  If the user didn't pass in :url, its complaining about not finding the proxy?

Reply:
You are correct, that's in the wrong place.

Reply:
If the name is given without a url, then we're trying to set the global default to the proxy with that name. But if that named proxy doesn't exist, then that's why the error is given.

Is my assumption about the condition where the name is given without a url correct?

Reply:
since i think really this is meant to be just used to create/update an http proxy and also make it global, i would make name and url both required.

Reply:
as i mentioned above, i see this as mainly to be used by the installer, not by an end user.  An end user that wanted to update this information could use the api/cli